apps/api/e2e/utils/api-client.ts

```diff
@@ -1276,4 +1279,64 @@ export class ApiClient {
       return this.handleApiError(error, "get user agent");
     }
   }
+
+  // ===========================
+  // Vote-related methods
+  // ===========================
+
+  /**
+   * Cast a vote for an agent in a competition
+   * Requires SIWE session authentication
+   */
+  async castVote(
+    agentId: string,
+    competitionId: string,
+  ): Promise<VoteResponse | ErrorResponse> {
+    try {
+      const response = await this.axiosInstance.post("/api/user/vote", {
+        agentId,
+        competitionId,
+      });
+      return response.data;
+    } catch (error) {
+      return this.handleApiError(error, "cast vote");
+    }
+  }
+
+  /**
+   * Get user's votes (optionally filtered by competition)
+   * Requires SIWE session authentication
+   */
+  async getUserVotes(
+    competitionId?: string,
+  ): Promise<UserVotesResponse | ErrorResponse> {
+    try {
+      const queryParams = competitionId
+        ? `?competitionId=${encodeURIComponent(competitionId)}`
+        : "";
+      const response = await this.axiosInstance.get(
+        `/api/user/votes${queryParams}`,
+      );
+      return response.data;
+    } catch (error) {
+      return this.handleApiError(error, "get user votes");
+    }
+  }
+
+  /**
+   * Get voting state for a user in a specific competition
+   * Requires SIWE session authentication
+   */
+  async getVotingState(
+    competitionId: string,
+  ): Promise<VotingStateResponse | ErrorResponse> {
+    try {
+      const response = await this.axiosInstance.get(
+        `/api/user/voting-state/${encodeURIComponent(competitionId)}`,
```

Comment:
maybe we can just call this: `GET /api/user/votes/:competitionId`

since it'll get voting information relative to a specific competition, irrespective of if you've voted or not

Reply:
actually...since the `/votes` url above uses `competitionId` as a query param, this would lead to confusion. maybe it should be: `/user/votes/:competitionId/state`